test/vinyl-luatest/gh_10603_pk_alter_vs_dml_test.lua

```diff
@@ -0,0 +1,91 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all(function(cg)
+    t.tarantool.skip_if_not_debug()
+    cg.server = server:new()
+    cg.server:start()
+end)
+
+g.after_all(function(cg)
+    cg.server:drop()
+end)
+
+g.after_each(function(cg)
+    cg.server:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        if box.space.test ~= nil then
+            box.space.test:drop()
+        end
+    end)
+end)
+
+g.test_empty_pk_alter = function(cg)
+    cg.server:exec(function()
+        local fiber = require('fiber')
+        local timeout = 10 -- seconds
+
+        local s = box.schema.space.create('test', {engine = 'vinyl'})
+        s:create_index('pk')
+
+        local c1 = fiber.channel(1)
+        local f1 = fiber.new(function()
+            s:replace({1, 10})
```

Comment:
Nit. Let's read from `c1` or let's get rid of it. Now it is redundant.

Reply:
Missed that. Fixed, thanks!

<details><summary>diff</summary>

```diff
diff --git a/test/vinyl-luatest/gh_10603_pk_alter_vs_dml_test.lua b/test/vinyl-luatest/gh_10603_pk_alter_vs_dml_test.lua
index 1b37f68f90dd..a6fdf28d88a2 100644
--- a/test/vinyl-luatest/gh_10603_pk_alter_vs_dml_test.lua
+++ b/test/vinyl-luatest/gh_10603_pk_alter_vs_dml_test.lua
@@ -32,6 +32,7 @@ g.test_empty_pk_alter = function(cg)
 
         local c1 = fiber.channel(1)
         local f1 = fiber.new(function()
+            c1:get()
             s:replace({1, 10})
         end)
         f1:set_joinable(true)
```

</details>